### PR TITLE
[release/6.0 Cleanup remaining SSP references

### PR DESCRIPTION
**Pull request by dougbu**
- follow up to e58a6c2
- remove references to System.Security.Permissions or its closure
  - only mentions are in eng/PackageOverrides.txt and eng/PlatformManifest.txt
  - those files will remain unused until we update them in the run-up to 6.0.1
- still indirectly pick up package refs for SSP and its closure but that does not impact targeting pack content

nit: Only need to exclude System.Net.Quic and System.Security.Cryptography.Pkcs from `@(_AvailableRuntimeRefAssemblies)`
  - assemblies are not in `@(ReferencePathWithRefAssemblies)`, just the transport package

### CODE REVIEW

**Comment by wtgodbe** [comment]
> those files will remain unused until we update them in the run-up to 6.0.1

Can you file an issue for this?

**Review by wtgodbe — APPROVED** [review]
Again LGTM assuming the outputs look good

**Comment by ViktorHofer** [comment]
> and System.Security.Cryptography.Pkcs from @(_AvailableRuntimeRefAssemblies)

How is Pkcs getting into the closure? Is it part of the transport pack?

**Comment by dougbu** [comment]
> Can you file an issue for this?

No need. The 6.0.1 build won't work w/o the updated files 😀 This is part of what we should document for major version branding updates too.

> How is Pkcs getting into the closure? Is it part of the transport pack?

We build the _entire_ closure now that 0b0bed3eb89a is in.

My point was different in any case: `@(_AvailableRuntimeRefAssemblies)` lists the files from the internal transport package. We need the ref/ assemblies for QUIC and PKCS to build our stuff but they must not land in our targeting package.

**Comment by dougbu** [comment]
``` text
Assert.Empty() Failure
Collection: ["System.Net.Quic", "System.Security.Cryptography.Pkcs"]
```

Need to double-check my `@(_AspNetCoreAppPackageOverrides)` exclusions and see where these assemblies are coming from. Likely need to undo a part of this change.

File: src/Framework/App.Ref/src/Microsoft.AspNetCore.App.Ref.csproj
Comment (ericstj) [inline src/Framework/App.Ref/src/Microsoft.AspNetCore.App.Ref.csproj:140]:
```suggestion
            @(ReferencePathWithRefAssemblies->WithMetadataValue('Filename', 'System.Net.Quic'));
            @(ReferencePathWithRefAssemblies->WithMetadataValue('Filename', 'System.Security.Cryptography.Pkcs'));
            @(ReferencePathWithRefAssemblies->WithMetadataValue('NuGetPackageId', 'Microsoft.NETCore.App.Ref'))" />
```

Comment (ericstj) [reply src/Framework/App.Ref/src/Microsoft.AspNetCore.App.Ref.csproj:140]: I see, I wasn't certain what any of this does, just noticed a place where I thought it might have been accidental.

**Comment by ericstj** [comment]
> How is Pkcs getting into the closure? Is it part of the transport pack?

It's a dependency of [Crypto.Xml](https://github.com/dotnet/runtime/blob/d801ae267dc2ea1b177f84af4736d71f4c94d88e/src/libraries/System.Security.Cryptography.Xml/src/System.Security.Cryptography.Xml.csproj#L134) which gets referenced by [DataProtection](https://github.com/dotnet/aspnetcore/blob/18215f9061928235775a9f6bd97e3f1da82fc9c4/src/DataProtection/DataProtection/src/Microsoft.AspNetCore.DataProtection.csproj#L29).  It's not exposed by any surface area in ASP.NET nor Crypto.Xml which is why it has been be excluded, but if ASP.NET wanted to implement the intent of your changes here https://github.com/dotnet/runtime/issues/53892, @ViktorHofer, they could decide to just expose it.  Now is probably not the time to make that change, but something to consider for 7.0 if it simplifies things.  We'd want to get @bartonjs's blessing that exposing Pkcs would be no worse than Crypto.Xml.

**Comment by bartonjs** [comment]
> We'd want to get @bartonjs's blessing that exposing Pkcs would be no worse than Crypto.Xml.

Pkcs we like.  We may not have the best of APIs in there, but it's based on tech we're happy with.  It's also a library that has seen new types and new functionality added to it since the start of .NET Core, unlike Crypto.Xml.

So, definitely "no worse than" :)

**Comment by dougbu** [comment]
> > We'd want to get @bartonjs's blessing that exposing Pkcs would be no worse than Crypto.Xml.

> Pkcs we like.  We may not have the best of APIs in there, but it's based on tech we're happy with.  It's also a library that has seen new types and new functionality added to it since the start of .NET Core, unlike Crypto.Xml.

> So, definitely "no worse than" :)

For 6.0, we ship both Crypto.Pkcs and Crypto.Xml in our shared frameworks but only Crypto.Xml in our targeting packs. Only the native aspnetcorev2_inprocess module (on Windows) and System.Diagnostics.EventLog.Messages (which isn't needed in the targeting pack) are treated similarly to Crypto.Pkcs.  So, Crypto.Pkcs is the one "obviously unaligned" assembly here.

We can align things by removing the Crypto.Pkcs exclusion. But, is this worthwhile for our 7.0.x customers if we aren't using Crypto.Pkcs types in our public API❔ If yes, we should first get Crypto.Pkcs into the Microsoft.Internal.Runtime.AspNetCore.Transport package.

**Pull request merged**
